azext_iot/central/commands_user.py

```diff
@@ -36,15 +37,50 @@ def add_user(
         return provider.add_email(
             assignee=assignee,
             email=email,
-            role=Role[role],
+            org_id=org_id,
+            role=role,
             central_dns_suffix=central_dns_suffix,
         )
 
     return provider.add_service_principal(
         assignee=assignee,
+        org_id=org_id,
         tenant_id=tenant_id,
         object_id=object_id,
-        role=Role[role],
+        role=role,
+        central_dns_suffix=central_dns_suffix,
+    )
+
+
+def update_user(
+    cmd,
+    app_id: str,
+    assignee: str,
+    roles=None,
+    email=None,
+    tenant_id=None,
+    object_id=None,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1.value,
+) -> UserType:
+    provider = CentralUserProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    if email:
+        return provider.update_email_user(
```

Comment:
Would there be a case where you would want to update the email and the service principal (ex: the update_user arguments include a new email and object_id)

Reply:
"email user" and "service principal user" are two different concept in central. if email is specified, other arguments are simply ignored.

Reply:
Can you add to the email param description which arguments would be ignored if the email is specified?

Reply:
added